src/m365/teams/commands/chat/chat-get.ts

```diff
@@ -0,0 +1,200 @@
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if (!args.options.id && !args.options.participants && !args.options.name) {
+      return 'Specify id or participants or name, one is required.';
+    }
+
+    let nrOfMutuallyExclusiveOptionsInUse = 0;
+    if (args.options.id) { nrOfMutuallyExclusiveOptionsInUse++; }
+    if (args.options.participants) { nrOfMutuallyExclusiveOptionsInUse++; }
+    if (args.options.name) { nrOfMutuallyExclusiveOptionsInUse++; }
+
+    if (nrOfMutuallyExclusiveOptionsInUse > 1) {
+      return 'Specify either id or participants or name, but not multiple.';
+    }
+
+    if (args.options.id && !validation.isValidTeamsChatId(args.options.id)) {
+      return `${args.options.id} is not a valid Teams ChatId.`;
+    }
+
+    if (args.options.participants) {
+      const participants = args.options.participants.toLowerCase().replace(/\s/g, '').split(',').filter(e => e && e !== '');
+      if (!participants || participants.length === 0 || participants.some(e => !validation.isValidUserPrincipalName(e))) {
+        return `${args.options.participants} contains one or more invalid email addresses.`;
+      }
+    }
+
+    return true;
+  }
+  
+  private async getChatDetailsById(id: string): Promise<Chat> {
+    const requestOptions: AxiosRequestConfig = {
+      url: `${this.resource}/v1.0/chats/${encodeURIComponent(id)}`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'      
+    };
+
+    const chat = await request.get<Chat>(requestOptions);
+    return chat;
+  }
+
+  private async getChatIdByParticipants(participantsString: string): Promise<string> {
+    const participants = participantsString.toLowerCase().replace(/\s/g, '').split(',').filter(e => e && e !== '');
+    const currentUserEmail = accessToken.getUserNameFromAccessToken(Auth.service.accessTokens[this.resource].accessToken).toLowerCase();
+    const existingChats = await this.findExistingChatsByParticipants([currentUserEmail, ...participants]);
+    
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with these participants.');
+    }
+
+    if (existingChats.length === 1) {
+      return existingChats[0].id as string;
+    }
+
+    const disambiguationText = existingChats.map(c => {
+      return `- ${c.id}${c.topic && ' - '}${c.topic} - ${c.createdDateTime && new Date(c.createdDateTime).toLocaleString()}`;
+    }).join(os.EOL);
+
+    throw new Error(`Multiple chat conversations with this topic found. Please disambiguate:${os.EOL}${disambiguationText}`);
+    
+  }
+  
+  private async getChatIdByName(name: string): Promise<string> {
+    const existingChats = await this.findExistingGroupChatsByTopic(name);
+
+    if (!existingChats || existingChats.length === 0) {
+      throw new Error('No chat conversation was found with this name.');
```

Comment:
We use `topic` and `name` in different places. Let's decide on one and use it consistently throughout

Reply:
Good point, I've changed it to `name`. Topic is now only used on the Microsoft Graph types and endpoints, but that's okay I think.